mkosxinstallusb.sh

```diff
@@ -107,30 +112,30 @@ fi
 
 echo "\n# converting installer disk image to raw format"
 dmg2img "$installer_dmg" InstallESD.img
-kpartx -s -a InstallESD.img
+InstallESD_partitions="$(map_partitions "InstallESD.img")"
 
 echo "\n# converting base system disk image to raw format"
 mkdir -p $InstallESD_mount_point
-mount /dev/mapper/loop0p2 $InstallESD_mount_point
+mount "${InstallESD_partitions}p2" $InstallESD_mount_point
 dmg2img $InstallESD_mount_point/BaseSystem.dmg BaseSystem.img
-kpartx -s -a BaseSystem.img
+BaseSystem_partitions="$(map_partitions "BaseSystem.img")"
 
 echo "\n# partitioning USB drive"
-umount $stick_dev? || true
+ls -1 "${stick_dev}"? | xargs -n1 umount || true
 sgdisk -o $stick_dev
 partprobe $stick_dev
 sgdisk -n 1:0:0 -t 1:AF00 -c 1:"disk image" -A 1:set:2 $stick_dev
-partprobe $stick_dev
```

Comment:
`sgdisk` in the line above re-partitions the USB stick to contain the single HFS+ partition. Without `partprobe` the kernel would still see the old partition table, so formatting can do any kind of damage.

Reply:
Are you sure on this? The reason why I removed it (and also the reason why I don't use `partprobe` everywhere in lieu of `kpartx` is that in my experience `kpartx` is a bit more standardized.
Reasons for using only `kaprtx` include:
* `kpartx` always seems to put the partitions into `/dev/mapper/{disk}p{partnum}`
* I'm not sure sure all distros create partitions for loop devices with `partprobe` (and, for the ones that do, they won't necessarily put them in `/dev/mapper`.
* In the past, I've experienced loop devices refusing to detach if partitions from `kpartx` haven't been unmapped (I think this is fixed in newer kernels and/or versions of `kpartx`). As a result, a concern of mine was that if `partprobe` does map these partitions (and if say it puts them in `/dev` and not `/dev/mapper`, the user would have issues detaching the loop device. I generally wasn't sure about the mechanics here, so I figured I'd be safe.
* I was under the impression that `kpartx` re-reads the partition table as does `partprobe`. Whenever I edit a partition table with `fdisk` and that utility can't update the partition table, it tells you to run `partprobe` *or* `kpartx`.

Reply:
I have checked on some old Ubuntu 14 instance and can confirm that `partprobe` is really not needed. Some years ago kernel did not re-read new partition table after making changes, so that required restarting or running `partprobe`. But now new partitions are visible right after the `sgdisk` run, so this call can be removed.